RetailCoder.VBE/API/ParserState.cs

```diff
@@ -28,30 +77,28 @@ public interface IParserState
     [ComVisible(true)]
     [Guid("3D8EAA28-8983-44D5-83AF-2EEC4C363079")]
     [InterfaceType(ComInterfaceType.InterfaceIsIDispatch)]
-    public interface IParserStateEvents
+    public interface IRubberduckParserStateEvents
     {
-        void OnParsed();
-        void OnReady();
-        void OnError();
+        void OnStateChanged(ParserState state);
```

Comment:
This is API design; the events aren't for us to consume, they're for VBA code to handle if the API user code wants to. A richer API is easier to code against: if every single line of VBA code ever written against that API must `Select Case state` and branch accordingly (*ideally* in separate methods, for separate concerns), then our API pretty much encourages breaking SRP and writing a `Select Case` or worse, a huge `If...Else If...Else` block to handle everything in one place.

On the other hand, if the API exposes a set of events to implement handlers for, you get to write your VBA code against a `WithEvents` local variable and cherry-pick which parser events you want to handle, and then the IDE generates a dedicated method stub for you. VBA can be object-oriented, but is first and foremost an *event-driven* language.

For example, an `ADODB.Recordset` exposes `FetchComplete`, `FetchProgress` and `FieldChangeComplete` events, but also `MoveComplete`, `RecordChangeComplete`, `RecordsetChangeComplete`, and then `WillChangeField`, `WillChangeRecord`, `WillChangeRecordset`, and `WillMove`; an `ADODB.Connection` object exposes `BeginTransComplete`, `CommitTransComplete`, `ConnectComplete`, `Disconnect`, `ExecuteComplete`, `InfoMessage`, `RollbackTransComplete`, and then `WillConnect` and `WillExecute` - the API fires events for everything potentially interesting that happens; extension points if you will: black-box code is running, but as it does so it can give you parameters and code to run before (or after) a number of things are done.

Every Office application offers a pretty rich event-driven API, too. Just look at workbook and spreadsheet events, or not even further than UserForms and controls, for more examples: events are *everywhere* in VBA.

The threading issue is real, the events somehow aren't fired from the thread they were registered from. See [this SO answer](http://stackoverflow.com/a/36266828/1188513) - I've tested a dumbed-down API, and it works nicely.